### PR TITLE
[15.0][MIG] stock_product_pack: Migration to version 15.0

### PR DESCRIPTION
**Pull request by ernesto-garcia-tecnativa**
@tecnativa TT36997

### CODE REVIEW

**Comment by ernesto-garcia-tecnativa** [comment]
@pedrobaeza @victoralmau please review

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -33,14 +33,14 @@ repos:
         language: fail
         files: '[a-zA-Z0-9_]*/i18n/en\.po$'
   - repo: https://github.com/oca/maintainer-tools
-    rev: e3b46970af62cbbf99f0b45a6675eb4cae744d6d
+    rev: dfba427ba03900b69e0a7f2c65890dc48921d36a
```
Comment (chienandalu) [inline .pre-commit-config.yaml]: Looks like you mixed up copier update changes with the migration commit

Comment (ernesto-garcia-tecnativa) [reply .pre-commit-config.yaml]: yes, I made a mistake but later I did a rebase to fix

Comment (pedrobaeza) [reply .pre-commit-config.yaml]: The problem is that such changes are part of the migration commit, so you should undo that.

Comment (ernesto-garcia-tecnativa) [reply .pre-commit-config.yaml]: I working in this, sorry

Comment (ernesto-garcia-tecnativa) [reply .pre-commit-config.yaml]: Done please review

File: stock_product_pack/views/product_template_views.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <!-- Generic Product Template view modification -->
+    <record id="product_template_form_view" model="ir.ui.view">
+        <field name="name">product.template.pack.form</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product_pack.product_template_form_view" />
+        <field name="arch" type="xml">
+            <field name="pack_modifiable" position="after">
+                <field
+                    name="dont_create_move"
+                    attrs="{'invisible':['|', ('pack_ok', '=', False), ('type', '=', 'service')]}"
```
Comment (victoralmau) [inline stock_product_pack/views/product_template_views.xml]:
```suggestion
                    attrs="{'invisible':['|', ('pack_ok', '=', False), ('detailed_type', '=', 'service')]}"
```

File: stock_product_pack/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        res = super()._compute_quantities_dict(
+            lot_id, owner_id, package_id, from_date=from_date, to_date=to_date
+        )
+        packs = self.filtered("pack_ok")
+        for product in packs.with_context(prefetch_fields=False):
+            pack_qty_available = []
+            pack_virtual_available = []
+            pack_free_qty = []
+            subproducts = product.pack_line_ids.filtered(
+                lambda p: p.product_id.type == "product"
```
Comment (victoralmau) [inline stock_product_pack/models/product_product.py]:
```suggestion
                lambda p: p.product_id.detailed_type == "product"
```

File: stock_product_pack/models/product_product.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            }
+        return res
+
+    def _compute_quantities(self):
+        """In v13 Odoo introduces a filter for products not services.
+        To keep how it was working on v12 we try to get stock for
+        service products if they are pack.
+        """
+        service_pack_products = self.filtered(
+            lambda p: p.type == "service" and p.pack_ok
```
Comment (victoralmau) [inline stock_product_pack/models/product_product.py]:
```suggestion
            lambda p: p.detailed_type == "service" and p.pack_ok
```

File: stock_product_pack/tests/test_stock_product_pack.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                "route_id": cls.env.ref("stock.route_warehouse0_mto").id,
+                "procure_method": "make_to_stock",
+                "warehouse_id": warehouse.id,
+                "location_id": cls.env.ref("stock.stock_location_stock").id,
+            }
+        )
+        cls.pack_dc_with_dm = cls.product_obj.create(
+            {
+                "name": "Pack With storeable and not move product",
+                "type": "product",
```
Comment (victoralmau) [inline stock_product_pack/tests/test_stock_product_pack.py]:
```suggestion
                "detailed_type": "product",
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by ernesto-garcia-tecnativa** [comment]
I pushed changes, please review

**Review by victoralmau — APPROVED** [review]
Code review OK

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**